// Generated from scheduler_sync_envelope.ifex by covesa-ifex-core
// DO NOT EDIT - regenerate with: ./generate_proto.sh
syntax = "proto3";



package swdv.scheduler_sync_envelope;

// Namespace: scheduler_sync
// Scheduler state synchronization types

// Types of sync events
enum sync_event_type_t {
  FULL_SYNC = 0;
  JOB_CREATED = 1;
  JOB_UPDATED = 2;
  JOB_DELETED = 3;
  JOB_EXECUTED = 4;
  HEARTBEAT = 5;
}

// Job status for sync (mirrors scheduler status)
enum job_sync_status_t {
  PENDING = 0;
  RUNNING = 1;
  COMPLETED = 2;
  FAILED = 3;
  CANCELLED = 4;
}

// Vehicle wake behavior for scheduled jobs
enum wake_policy_t {
  NO_WAKE = 0;
  WAKE_REQUIRED = 1;
}

// Vehicle sleep behavior during job execution
enum sleep_policy_t {
  SLEEP_NORMAL = 0;
  INHIBIT_UNTIL_COMPLETE = 1;
}

// Full job information for sync
message job_info_t {
  string job_id = 1;
  string title = 2;
  string service = 3;
  string method = 4;
  string parameters = 5;
  string scheduled_time = 6;
  string recurrence_rule = 7;
  string next_run_time = 8;
  job_sync_status_t status = 9;
  wake_policy_t wake_policy = 10;
  sleep_policy_t sleep_policy = 11;
  uint32 wake_lead_time_s = 12;
  uint64 created_at_ms = 13;
  uint64 updated_at_ms = 14;
}

// Result of a job execution (for JOB_EXECUTED events)
message execution_result_t {
  string job_id = 1;
  job_sync_status_t status = 2;
  uint64 executed_at_ms = 3;
  uint32 duration_ms = 4;
  string result = 5;
  string error_message = 6;
  string next_run_time = 7;
}

// A single sync event
message sync_event_t {
  sync_event_type_t event_type = 1;
  uint64 sequence_number = 2;
  uint64 timestamp_ns = 3;
  string job_id = 4;
  job_info_t job_info = 5;
  execution_result_t execution_result = 6;
}

// Sync message containing one or more events
message sync_message_t {
  string vehicle_id = 1;
  string bridge_instance_id = 2;
  repeated sync_event_t events = 3;
  uint32 active_jobs_count = 4;
  uint32 state_checksum = 5;
}

// Cloud acknowledgment of sync message
message sync_ack_t {
  uint64 last_sequence_received = 1;
  bool checksum_match = 2;
  bool request_full_sync = 3;
}


